File: engage-4 (Angular en Spring)/engageserver/src/main/java/app/models/user/UserReview.java
package app.models.user;

import javax.persistence.*;

@Entity
public class UserReview {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    @ManyToOne
    private User evaluator;
    @ManyToOne
    private UserAttribute evaluated;
    private int rating;
    private long time;

    public UserReview() {

    }

    public UserReview(User evaluator, UserAttribute evaluated, int rating, long time) {
        this.evaluator = evaluator;
        this.evaluated = evaluated;
        this.rating = rating;
        this.time = time;
    }

    public long getId() {
        return id;
    }

    public User getEvaluator() {
        return evaluator;
    }

    public UserAttribute getEvaluated() {
        return evaluated;
    }

    public int getRating() {
        return rating;
    }

    public long getTime() {
        return time;
    }

    @Override
    public String toString() {
        return "UserReview{" +
                "id=" + id +
                ", evaluator=" + evaluator +
                ", evaluated=" + evaluated +
                ", rating=" + rating +
                ", time=" + time +
                '}';
    }
}

File: engage-4 (Angular en Spring)/engage/src/app/models/user/user-review.ts
import {UserAttribute} from './user-attribute';
import {User} from './user.model';

export class UserReview {
  id: number;
  evaluator: User;
  evaluated: UserAttribute;
  rating: number;
  time: number;

  constructor(evaluator: User, evaluated: UserAttribute, rating: number, time: number) {
    this.evaluator = evaluator;
    this.evaluated = evaluated;
    this.rating = rating;
    this.time = time;
  }
}